### ISSUE TITLE
Accessibility Review: Iowa Home Page Brand Bar / Search / Menu

### ISSUE THREAD

**Issue by bspeare** [uiowa/uids#55]
Url to review: https://uiowa.dev.drupal.uiowa.edu/

Todd's notes:

### ARIA Landmark Roles
Per SiteImprove’s assessment algorithm, landmark regions and other roles with native HTML5 equivalents no longer require the redundancy of the role attribute. 
Containers that use HTML5 and ARIA landmarks redundantly will generate a warning in SiteImprove. WAVE, AInspector, and aXe do not recognize this as an error or warning. Unable to determine whether this applies to elements other than landmarks that have both native and ARIA roles. Multiple instances of landmarks or other roles must still include accessible names.

### University of Iowa Brand Bar with Search and Menu

#search-wrapper
- [x] Apply role=”search” to search-wrapper to announce landmark?
- [x] Accessible name changes when expanded/collapsed. Recommend “SEARCH” name remain in place for “expanded/collapsed” state change—similar to accordion
- [x] When SEARCH region is expanded, user must SHIFT+TAB to focus to Search field—move focus directly there?
- [x] “Search button” is used as name for two elements. Could search form button be “Submit Search” or similar?

div.o-canvas__group (First instance)
- [ ] Apply unique ID to distinguish instance of class
- [x] Apply role=”navigation” to announce landmark
- [x] Apply accessible name to above navigation role

a.sf-depth-1 menuparent ()
- [ ] Can these use role=”button”?
- [ ] aria-haspopup=”true”
- [ ] aria-expanded (toggle)
- [x] User becomes lost in sub-menu—keyboard pattern?
- [ ] Keyboard pattern? Can these elements be focusable by tab; child elements by directionals?

**Comment by bspeare** [comment]
@inspector508 Can you review the checked off items listed above on https://uiowa.stage.drupal.uiowa.edu/?   

A few notes/questions:

**#search-wrapper**

- I did apply the role="search" to .search-wrapper but the form within the wrapper also has role="search" - is that too much?

**div.o-canvas__group (First instance)**

- This structure has changed a bit since your first review.  Can you look again at https://uiowa.stage.drupal.uiowa.edu/ and let me know if still is still necessary?

**a.sf-depth-1 menuparent ()**

- The unchecked items under "a.sf-depth-1 menuparent ()" will require a greater amount of investment writing custom JavaScript to override the Drupal module.   Do the unchecked items also apply to the dropdown implementation on https://hr.uiowa.edu/ or is it just specific to this implementation?

**Comment by richardbporter** [comment]
I was looking for an issue related to the website feedback we received suggesting we auto-focus the text input after clicking search. This issue seems to reference it but it is checked off. Is this still something we'd to do?

**Comment by bspeare** [comment]
@richardbporter we should probably close this issue because @joewhitsitt worked on adjusting the the search bar based on Todd's requests and I think that involved removing the search focus because it created a11y issues: https://github.com/uiowa/uids/pull/241.
